packages/code-du-travail-frontend/src/layout/ThemeProvider.js

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useContext, useEffect, useMemo } from "react";
+import { ThemeProvider as StyledThemeProvider } from "styled-components";
+
+import { theme } from "@socialgouv/react-ui";
+
+const HIGH_CONTRAST_STORAGE_KEY = "highContrast";
+
+const { colors, highContrastColors } = theme;
+
+const ThemeContext = React.createContext({
+  currentTheme: colors,
+  toggleTheme: () => {}
+});
+
+export const ThemeProvider = props => {
+  const [currentTheme, setCurrentTheme] = useState(colors);
+
+  useEffect(() => {
+    localStorage.getItem(HIGH_CONTRAST_STORAGE_KEY) === "true" &&
```

Comment:
there is useLocalStorageHook :)

Reply:
I'm going to take a look !

Reply:
Bon, je l'ai utilisé. Je ne suis pas sur d'avoir simplifié le code mais je l'ai utilisé !